db/db_impl/db_impl.h

```diff
@@ -1240,6 +1240,38 @@ class DBImpl : public DB {
 
   std::atomic<bool> shutting_down_;
 
+  class RecoveryVersionEdits {
```

Comment:
This is not just version edits. We can come up with another name, VersionEditsContext, etc?